### PR TITLE
RHCLOUD-36073 limit workspace queryset to requester tenant

### PR DESCRIPTION
**Pull request by coderbydesign**
## Link(s) to Jira
- https://issues.redhat.com/browse/RHCLOUD-36073

## Description of Intent of Change(s)
We'll eventually have meta-authz on cross-org workspace access enforcement, but for now we should limit any v2 viewsets' querysets to that of the tenant for the requester.

In order to ensure we're starting with a default queryset that restricts a v2
model's API response to only records for the requester's tenant, this abstracts
an override of the `get_queryset` method on our existing `BaseV2ViewSet` class
which v2 viewsets inherit from.

We can dynamically set the queryset model by adding a `basename` value [1] which
the queryset uses, filtering to the request tenant.

[1] https://www.django-rest-framework.org/api-guide/routers/

## Local Testing
As a user of org1 I should only see workspaces for org1 when I query GET /api/rbac/v2/workspaces/

## Checklist
- [ ] if API spec changes are required, is the spec updated?
- [ ] are there any pre/post merge actions required? if so, document here.
- [ ] are theses changes covered by unit tests?
- [ ] if warranted, are documentation changes accounted for?
- [ ] does this require migration changes?
  - [ ] if yes, are they backwards compatible?
- [ ] is there known, direct impact to dependent teams/components?
  - [ ] if yes, how will this be handled?

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Practices Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by astrozzc — APPROVED** [review]
LGTM!

File: rbac/management/workspace/view.py [hunk 1]
```diff
@@ -99,7 +97,8 @@ def update(self, request, *args, **kwargs):
 
     def partial_update(self, request, *args, **kwargs):
         """Patch a workspace."""
-        payload = json.loads(request.body or "{}")
+        self.validate_workspace(request, "patch")
+        payload = request.data or "{}"
         for field in payload:
```
Comment (alechenninger) [inline rbac/management/workspace/view.py:102]: Should the other end of `or` still be a string `"{}"` or should it be a dictionary literal `{}`?

Comment (alechenninger) [reply rbac/management/workspace/view.py:102]: Looks like it should be a `{}` if I understand the intent. Confirmed with this test:

```
    def test_partial_update_empty(self):
        """Test for updating a workspace with empty body."""
        workspace_data = {
            "name": "New Workspace",
            "description": "New Workspace - description",
            "tenant_id": self.tenant.id,
        }

        workspace = Workspace.objects.create(**workspace_data)

        url = reverse("v2_management:workspace-detail", kwargs={"pk": workspace.id})
        client = APIClient()
        response = client.patch(url, {}, format="json", **self.headers)
        self.assertEqual(response.status_code, status.HTTP_200_OK)
        self.assertEqual(response.get("content-type"), "application/json")
        data = response.data
        self.assertEqual(data.get("name"), "New Workspace")
        self.assertEqual(data.get("description"), "New Workspace - description")
        self.assertEqual(data.get("id"), str(workspace.id))
        self.assertNotEquals(data.get("created"), "")
        self.assertNotEquals(data.get("modified"), "")
        self.assertEquals(data.get("type"), "standard")

        update_workspace = Workspace.objects.filter(id=workspace.id).first()
        self.assertEquals(update_workspace.name, "New Workspace")
```

Comment (coderbydesign) [reply rbac/management/workspace/view.py:102]: Good catch, thanks! Updated w/test.

**Review by lpichler — APPROVED** [review]

**Pull request merged**